listenbrainz_spark/recommendations/recording/user_similarity.py

```diff
@@ -0,0 +1,98 @@
+import sys
+import math
+
+from flask import current_app
+from pyspark.mllib.linalg.distributed import CoordinateMatrix, MatrixEntry
+from pyspark.ml.stat import Correlation
+from pyspark.sql.functions import struct, collect_list
+
+import listenbrainz_spark
+from listenbrainz_spark import SparkSessionNotInitializedException, utils, path
+from listenbrainz_spark.exceptions import PathNotFoundException, FileNotFetchedException
+
+
+def create_messages(similar_users_df):
+    itr = similar_users_df.toLocalIterator()
+    message = {}
+    for row in itr:
+        message[row.user_name] = {user.other_user_name: user.similarity for user in row.similar_users}
+    yield {
+        'type': 'similar_users',
+        'data': message
+    }
+
+
+def threshold_similar_users(matrix, threshold):
```

Comment:
types and docstrings on these methods would really help readability.

Reply:
Done.